### PR TITLE
RHEL: fix Azure images configuration according to documentation

### PR DESCRIPTION
**Pull request by thozza**
Make sure that the RHEL-8 and RHEL-9 Azure images have the following configuration consistent with the official RHEL documentation:

- kernel commandline 
- modules disabled via modprobe configuration
- grub2 configuration in `/etc/default/grub`

Depends on https://github.com/osbuild/osbuild/pull/1601

Related to https://issues.redhat.com/browse/RHEL-19583

### CODE REVIEW

File: pkg/distro/rhel9/azure.go [hunk 1]
```diff
@@ -414,7 +414,7 @@ func azureRhuiBasePartitionTables(t *imageType) (disk.PartitionTable, bool) {
 	}
 }
 
-var defaultAzureKernelOptions = "ro console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300"
+var defaultAzureKernelOptions = "ro loglevel=3 console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300"
```
Comment (thozza) [inline pkg/distro/rhel9/azure.go:417]: I'm not 100% sure if adding `loglevel=3` is a good idea, but it is used in RHEL documentation and RHEL images built by MSFT in their marketplace have it set... :thinking:

File: pkg/distro/rhel8/azure.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ import (
 	"github.com/osbuild/images/pkg/subscription"
 )
 
-const defaultAzureKernelOptions = "ro crashkernel=auto console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300"
+const defaultAzureKernelOptions = "ro loglevel=3 crashkernel=auto console=tty1 console=ttyS0 earlyprintk=ttyS0 rootdelay=300"
```
[the remark this answers is not in the document]
Comment (mvo5) [reply pkg/distro/rhel8/azure.go:14]: Maybe worth a comment in the code then? something like: `// loglevel=3 as described in the RHEL documentation` or `// loglevel=3 to be consistent with existing MSFT images` maybe (or both :) ?

File: pkg/distro/rhel9/azure.go [hunk 3]
```diff
@@ -456,6 +456,12 @@ var defaultAzureImageConfig = &distro.ImageConfig{
 				osbuild.NewModprobeConfigCmdBlacklist("amdgpu"),
 			},
 		},
+		{
+			Filename: "blacklist-intel-cstate.conf",
```
Comment (mvo5) [inline pkg/distro/rhel9/azure.go:460]: Maybe a small comment in this codeblock that adds the blacklists that include the references from the commit message? (https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/deploying_rhel_9_on_microsoft_azure/assembly_deploying-a-rhel-image-as-a-virtual-machine-on-microsoft-azure_cloud-content-azure#making-configuration-changes_configure-the-image-azure) ? Or just somewhere in the file where we customize things (as it's more than just blacklisting that this describes)

**Review by mvo5 — APPROVED** [review]
Thanks, this looks fine, some inline comments but feel free to ignore.

File: pkg/distro/rhel9/azure.go (hunk 3 above)
Comment (thozza) [reply pkg/distro/rhel9/azure.go:462]: I added a comment at the beginning of the image config.

File: pkg/distro/rhel8/azure.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (thozza) [reply pkg/distro/rhel8/azure.go:15]: I added the comment.

**Review by ondrejbudai — APPROVED** [review]
:heart_eyes: Thanks!

**Pull request merged**